### PR TITLE
Refactor Scan

### PR DESCRIPTION
**Pull request by baude**
Refactor scan to use the backends as well as container and
image objects.

### CODE REVIEW

File: Atomic/scan.py [hunk 1]
```diff
@@ -178,50 +177,70 @@ def set_scanner():
...
+        #        docker_object = docker_object if docker_object is not None \
+        #            else (next((item for item in images if item['Id'] == self.get_input_id(scan_input)), None))
+        #        if docker_object is not None:
+        #            docker_object['input'] = scan_input
+        #            scan_list.append(docker_object)
+        #    if len(scan_list) < 1:
+        #        raise ValueError("You must provide at least one container or image for atomic "
+        #                         "scan. See 'atomic scan --help' for more information")
+        #
+        #return scan_list
```
Comment (giuseppe) [inline Atomic/scan.py]: let's drop completely this dead code

File: Atomic/scan.py [hunk 2]
```diff
@@ -178,50 +177,70 @@ def set_scanner():
...
-                    docker_object['input'] = scan_input
-                    scan_list.append(docker_object)
-            if len(scan_list) < 1:
-                raise ValueError("You must provide at least one container or image for atomic "
-                                 "scan. See 'atomic scan --help' for more information")
-
+            for scan_target in self.args.scan_targets:
+                try:
+                    # get_backend_and_container throws a ValueError when it cannot find anything
+                    _, scan_obj = beu.get_backend_and_container_obj(scan_target)
```
Comment (giuseppe) [inline Atomic/scan.py]: could we rewrite it as `_, scan_list = beu.get_backend_and_container_obj(scan_target)`

and move the second try/except block to the same level as the first one?  The only other change required is:
```
if scan_list != []:
   _, scan_list = beu.get_backend_and_image_obj(scan_target)
```

What do you think?

File: Atomic/scan.py (hunk 1 above)
Comment (baude) [inline Atomic/scan.py]: Missed that, of course.

**Comment by giuseppe** [comment]
@rh-atomic-bot r+ dc6def0

**Pull request closed without merging**